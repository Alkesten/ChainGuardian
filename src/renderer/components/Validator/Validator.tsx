import * as React from "react";
import {ButtonPrimary, ButtonDestructive} from "../Button/ButtonStandard";
import {AddButton} from "../Button/ButtonAction";
import {ValidatorStat} from "../Cards/ValidatorStat";
import {NodeCard} from "../Cards/NodeCard";
import {useState} from "react";

export interface IStatsType{
    roi: number;
    balance: bigint | number;
    uptime: number;
}
export interface IBeaconNode{
    id: string;
    url: string;
    respTime: number;
}
export interface IValidatorProps {
    name: string;
    stats: IStatsType;
    beaconNodes: IBeaconNode[];
    onBeaconNodeClick: () => void;
    onRemoveClick: () => void;
    onDetailsClick: () => void;
    onAddNodeClick: () => void;
}

export const Validator: React.FunctionComponent<IValidatorProps> = (
    props: IValidatorProps) => {
    const [nodeStatus, setNodeStatus] = useState(false);
    
    let numOfNodes = 0;
    for (let i = 0; i < props.beaconNodes.length; i++) {
        if (props.beaconNodes[i].id) numOfNodes++;
    }

    const getShowMore = (): void => {
        !nodeStatus ? setNodeStatus(true) : setNodeStatus(false);
    };

    const renderBeaconNodeCard = (x: IBeaconNode): React.ReactElement => {
        return(
            <NodeCard 
                onClick={(): void=>{props.onBeaconNodeClick;}}
                key={x.id} 
                value={x.respTime} 
                title={x.id}
                url={x.url}/>
        );
    };

    return(
        <div className="validator-container">
            <div className="validator-stats">
                <h2>Validator {props.name}</h2>
                <div className="validator-stats-container" >
                    <ValidatorStat value={props.stats.roi} title="Return (ETH)" type="ROI"/>
                    <ValidatorStat value={props.stats.balance} title="Balance" type="ETH"/>
                    <ValidatorStat value={props.stats.uptime} title="Uptime" type="Uptime"/>
                </div>
            </div>
            <div className="validator-nodes">
                <div className="node-container">
                    <div className="node-grid-container" >
                        {
                            nodeStatus ?
                                props.beaconNodes.map(node => {
                                    return renderBeaconNodeCard(node);
                                })
                                :
                                props.beaconNodes.filter((e)=>{
                                    if(props.beaconNodes.indexOf(e)<2){
                                        return e; 
                                    }
                                }).map(node => {
                                    return renderBeaconNodeCard(node);
                                })
                        }
                    </div>
                    <AddButton onClick={(): void=>{props.onAddNodeClick;}}/>
                    <div 
                        onClick={(): void=>{getShowMore();}} 
                        className={`show 
                            ${numOfNodes>2 ? "":"none"}
                            ${nodeStatus ? "less":"more"}`}>
                        {nodeStatus ? "Show Less":"Show More"}
                    </div>
                </div>
                <div className="validator-buttons">
                    <ButtonDestructive onClick={(): void=>{props.onRemoveClick;}}>REMOVE</ButtonDestructive>
                    <ButtonPrimary onClick={(): void=>{props.onDetailsClick;}}>DETAILS</ButtonPrimary>
                </div>
            </div>
        </div>
    );
};