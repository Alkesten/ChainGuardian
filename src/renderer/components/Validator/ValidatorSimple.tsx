import React, {useEffect} from "react";
import {useDispatch} from "react-redux";

import {loadValidatorBeaconNodes} from "../../actions/network";
import {BeaconNode} from "../../models/beaconNode";
import {ButtonSecondary, ButtonDestructive} from "../Button/ButtonStandard";
import {ValidatorStat} from "../Cards/ValidatorStat";
import {PrivateKeyField} from "../PrivateKeyField/PrivateKeyField";
import {InputForm} from "../Input/InputForm";
import {NodeCard} from "../Cards/NodeCard";

export interface IValidatorSimpleProps {
    name: string,
    status: string,
    publicKey: string,
    deposit: number,
    onRemoveClick: () => void;
    onExportClick: () => void;
    privateKey: string;
    nodes: BeaconNode[];
}

export const ValidatorSimple: React.FunctionComponent<IValidatorSimpleProps> = (
    props: IValidatorSimpleProps) => {
    const dispatch = useDispatch();

    useEffect(() => {
        const timeoutID = setInterval(() => {
            dispatch(loadValidatorBeaconNodes(props.publicKey));
        }, 5000);

        return () => clearInterval(timeoutID);
    }, [props.publicKey]);

    const renderBeaconNodes = (): React.ReactElement => {
        return (
            <div className="validator-nodes">
                <div className="node-container">
                    <div className="node-grid-container">
                        {props.nodes.map(node => {
                            return (
                                <NodeCard
                                    key={node.url}
                                    onClick={(): void => {
                                    }}
                                    title={node.localDockerId ? "Local Docker container" : "Remote Beacon node"}
                                    url={node.url}
                                    isSyncing={node.isSyncing}
                                    value={node.currentSlot || "N/A"}
                                />
                            );
                        })}
                    </div>
                </div>
            </div>
        );
    };

    return(
        <div className="validator-simple-container">
            <div className="validator-simple-keys">
                <h2>{props.name}</h2>
                <h3>Status: {props.status}</h3>
                <br />

                <ValidatorStat title="Deposit" type="ETH" value={props.deposit}/>

                <br />

                <InputForm
                    label="PUBLIC KEY"
                    focused={false}
                    inputValue={props.publicKey}
                    readOnly={true}
                    type="text"
                />

                <PrivateKeyField
                    label="PRIVATE KEY"
                    inputValue={props.privateKey}
                />
            </div>
            <div className="validator-simple-status">
                {renderBeaconNodes()}
                <div className="validator-simple-buttons">
                    <ButtonDestructive onClick={props.onRemoveClick}>REMOVE</ButtonDestructive>
                    <ButtonSecondary onClick={props.onExportClick}>EXPORT</ButtonSecondary>
                </div>
            </div>
        </div>
    );
};