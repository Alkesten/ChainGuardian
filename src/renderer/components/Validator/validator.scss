.validator-container{
    background-color: $neutral-700;
    box-shadow: 0px 2px 56px rgba(0, 0, 0, 0.3);
    border-radius: 24px;
    padding: 32px;
    width: 1280px;
    display: flex;
    justify-content: space-between;
}
.validator-stats-container{
    padding: 24px 0 0 8px;
    display: flex;
    .validator-card-container:nth-child(2){
        margin: 80px -20px 0px -20px ;
    }
}
.validator-nodes{
    display: flex;
    flex-direction: column;
    justify-content: center;
}
.node-container{
    padding: 24px;
    background-color: $neutral-800;
    box-shadow: inset 0px 2px 6px rgba(10, 39, 51, 0.2);
    border-radius: 16px;
    .add-button{
        position: fixed;
        margin-top: -40px;
        margin-left: 640px;
    }
}
.node-grid-container{
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-column-gap: 24px;
    grid-row-gap: 24px;
}
.node-card-container{
    &.none{
        display: none;
    }
}
.show{
    font-family: "Palanquin Dark", sans-serif;
    font-weight: 500;
    font-size: 20px;
    line-height: 22px;
    letter-spacing: 0.04em;
    color: $primary-blue-300;
    margin-left: 480px;
    width: 144px;
    margin-top: 15px;
    cursor: pointer;
    &.more{
        background: url(../assets/img/validator/show_more.svg) no-repeat;
        background-position: top 10px right 10px;
    }
    &.less{
        background: url(../assets/img/validator/show_less.svg) no-repeat;
        background-position: top 10px right 10px;
    }
    &.none{
        display: none;
    }
}
.validator-buttons{
    display: flex;
    justify-content: flex-end;
    padding-top: 32px;
    .button:nth-child(1){
        margin-right: 27px;
    }
}