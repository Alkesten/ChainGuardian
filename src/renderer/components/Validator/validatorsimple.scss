.validator-simple-container{
    background-color: $neutral-700;
    box-shadow: 0px 2px 56px rgba(0, 0, 0, 0.3);
    border-radius: 24px;
    padding: 32px;
    width: 1280px;
    display: flex;
    justify-content: space-between;
}
    .validator-simple-keys{
        // width: 50%;
        width: 800px;
        display: flex;
        flex-direction: column;
        justify-content: space-between;
    }
        .public-key{
            padding: 32px 0 0 10px;
        }

    .validator-simple-status{
        display: flex;
        flex-direction: column;
        align-items: flex-end;
        justify-content: space-between;
    }
        .validator-simple-status h2{
            margin-bottom: 25px;
            align-self: baseline;
        }
        .validator-simple-buttons{
            margin-top: 25px;
            display: flex;
            justify-content: flex-end;
            .button:nth-child(1){
                margin-right: 27px;
            }
        }