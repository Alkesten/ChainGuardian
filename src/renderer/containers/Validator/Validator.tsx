import React, {useState} from "react";
import {useSelector, useDispatch} from "react-redux";
import {useHistory} from "react-router";
import {startValidatorService, stopValidatorService} from "../../actions";
import {PasswordPrompt} from "../../components/Prompt/PasswordPrompt";
import {Routes} from "../../constants/routes";

import {IRootState} from "../../reducers";
import {calculateROI} from "../../services/utils/math";
import {AddButton} from "../../components/Button/ButtonAction";
import {ButtonDestructive, ButtonPrimary} from "../../components/Button/ButtonStandard";
import {ValidatorStat} from "../../components/Cards/ValidatorStat";
import {PrivateKeyField} from "../../components/PrivateKeyField/PrivateKeyField";
import {InputForm} from "../../components/Input/InputForm";
import {NodeCard} from "../../components/Cards/NodeCard";
import {Keypair} from "@chainsafe/bls";

export interface IValidatorSimpleProps {
    name: string,
    status: string,
    publicKey: string,
    onRemoveClick: () => void;
    onDetailsClick: () => void;
    onBeaconNodeClick: (id: string) => () => void;
}

export const Validator: React.FunctionComponent<IValidatorSimpleProps> = (
    props: IValidatorSimpleProps) => {
    const [askPassword, setAskPassword] = useState<string>(null);
    const dispatch = useDispatch();
    const history = useHistory();
    const validators = useSelector((state: IRootState) => state.validators);
    const network = useSelector((state: IRootState) => state.network.selected);
    const validatorBeaconNodes = useSelector((state: IRootState) => state.network.validatorBeaconNodes);
    const nodes = Object.prototype.hasOwnProperty.call(validatorBeaconNodes, props.publicKey) ?
        validatorBeaconNodes[props.publicKey] : [];

    const validator = validators[props.publicKey];
    const isLoaded = !!validator;
    const balance = isLoaded ? validator.balance || 0n : 0n;
    const ROI = calculateROI(balance, network);

    const onAddButtonClick = (): void => {
        history.push(Routes.ADD_BEACON_NODE.replace(":validatorKey", props.publicKey));
    };

    const renderBeaconNodes = (): React.ReactElement => {
        return (
            <div className="validator-nodes">
                <div className="box node-container">
                    <div className="node-grid-container">
                        {nodes.length === 0 ? <p>No working beacon nodes.</p> : null}

                        {nodes.map((node, index) => (
                            <NodeCard
                                key={index}
                                //TODO: change to some other id when multinode is enabled
                                onClick={props.onBeaconNodeClick(node.url)}
                                title={node.localDockerId ? "Local Docker container" : "Remote Beacon node"}
                                url={node.url}
                                isSyncing={node.isSyncing}
                                value={node.currentSlot || "N/A"}
                            />
                        ))}
                    </div>

                    <AddButton onClick={onAddButtonClick} />
                </div>
            </div>
        );
    };

    const controlValidator = async(keypair: Keypair): Promise<void> => {
        if (askPassword === "stop") {
            dispatch(stopValidatorService(keypair));
        } else if (askPassword === "start") {
            dispatch(startValidatorService(keypair));
        } else if (askPassword === "remove") {
            props.onRemoveClick();
        }
        setAskPassword(null);
    };

    const renderValidatorButtons = (): React.ReactElement => {
        const isRunning = validators[props.publicKey].isRunning;
        return (
            <div className="flex validator-service-button">
                {isRunning ?
                    <ButtonDestructive onClick={(): void => setAskPassword("stop")}>
                        Stop
                    </ButtonDestructive>
                    :
                    <ButtonPrimary onClick={(): void => setAskPassword("start")}>
                        Start
                    </ButtonPrimary>
                }
            </div>
        );
    };

    return(
        <>
            <div className="validator-container">
                <div className="validator-simple-keys">
                    <div className="row">
                        <h2>{props.name}</h2>
                        {renderValidatorButtons()}
                    </div>
                    <h3>Status: {validator.status || "N/A"}</h3>

                    <br />

                    <div className="row validator-stat-container ">
                        <ValidatorStat title="Balance" type="ETH" value={balance}/>
                        <ValidatorStat title="Return (ETH)" type="ROI" value={ROI}/>
                        <ValidatorStat title="Validator" type="Status" value={props.status}/>
                    </div>

                    <br />

                    <InputForm
                        label="PUBLIC KEY"
                        focused={false}
                        inputValue={props.publicKey}
                        readOnly={true}
                        type="text"
                    />

                    <PrivateKeyField
                        label="PRIVATE KEY"
                        keystore={validator.keystore}
                    />
                </div>
                <div className="validator-status">
                    {renderBeaconNodes()}
                    <div className="validator-buttons">
                        <ButtonDestructive onClick={(): void => setAskPassword("remove")}>REMOVE</ButtonDestructive>
                        <ButtonPrimary onClick={props.onDetailsClick}>DETAILS</ButtonPrimary>
                    </div>
                </div>
            </div>

            <PasswordPrompt
                keystore={validator.keystore}
                display={!!askPassword}
                onSubmit={controlValidator}
                onCancel={(): void=> setAskPassword(null)}
            />
        </>
    );
};
