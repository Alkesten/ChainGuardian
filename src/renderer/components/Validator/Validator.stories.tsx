import * as React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean, number, select, object, array } from '@storybook/addon-knobs';
import { Validator } from "./Validator";
import { ValidatorSimple } from "./ValidatorSimple";

storiesOf('Validator', module).add('Validator Component', () => {
    const stats = {
        roi: 10,
        balance: 0.1206,
        uptime: 92.1
    }    
    let node = [];
    const index = number("number of nodes",2);
    for (let i = 0; i < index; i++) {
        node.push({
            id: "BeaconNode",
            url: "rocketsonic.hr",
            respTime: 21
        })
    }
    const statsValue = object("stats",stats);
    const title = text("name","Validator 001");
    return <div >
            <Validator
            name={title}
            onBeaconNodeClick={()=>console.log("")} 
            onDetailsClick={()=>console.log("")} 
            onRemoveClick={()=>console.log("")} 
            onAddNodeClick={()=>console.log("")}
            stats={statsValue} 
            beaconNodes={node}/>
        </div>;
}).addDecorator(withKnobs);

storiesOf('Validator', module).add('Validator simple', () => {
    const depositValue = number("deposit", 0.3403);
    const title = text("name","Validator 001");
    const statusValue = text("status","Not working");
    const publicKeyValue = text("public key","6ffa3d24c9c26877d4a8bfa87455f44666ce93b7e13a3f84");
    return <div >
            <ValidatorSimple
            name={title}
            status={statusValue}
            publicKey={publicKeyValue}
            deposit={depositValue}
            onExportClick={()=>console.log("")} 
            onRemoveClick={()=>console.log("")} 
            />
        </div>;
}).addDecorator(withKnobs);