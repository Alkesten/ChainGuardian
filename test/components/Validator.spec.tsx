import * as React from "react";
import * as renderer from "react-test-renderer";
import {Validator} from "../../src/renderer/components/Validator/Validator";
// import {ValidatorSimple} from "../../src/renderer/components/Validator/ValidatorSimple";

describe("Validator", () => {
    it("renders correctly, 2 nodes", () => {
        const tree = renderer
            .create(<Validator
                name={"Test title"}
                onBeaconNodeClick={(): void=>{}} 
                onDetailsClick={(): void=>{}} 
                onRemoveClick={(): void=>{}} 
                onAddNodeClick={(): void=>{}}
                stats={{
                    roi: 10,
                    balance: 0.1206,
                    uptime: 92.1
                }} 
                beaconNodes={[{
                    id: "BeaconNode1",
                    url: "rocketsonic1.hr",
                    respTime: 21
                },
                {
                    id: "BeaconNode2",
                    url: "rocketsonic2.hr",
                    respTime: 22
                }]}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("renders correctly, 4 nodes", () => {
        const tree = renderer
            .create(<Validator
                name={"Test title"}
                onBeaconNodeClick={(): void=>{}} 
                onDetailsClick={(): void=>{}} 
                onRemoveClick={(): void=>{}} 
                onAddNodeClick={(): void=>{}}
                stats={{
                    roi: 10,
                    balance: 0.1206,
                    uptime: 92.1
                }} 
                beaconNodes={[{
                    id: "BeaconNode1",
                    url: "rocketsonic1.hr",
                    respTime: 21
                },
                {
                    id: "BeaconNode2",
                    url: "rocketsonic2.hr",
                    respTime: 22
                },
                {
                    id: "BeaconNode3",
                    url: "rocketsonic2.hr",
                    respTime: 22
                },
                {
                    id: "BeaconNode4",
                    url: "rocketsonic2.hr",
                    respTime: 22
                }]}/>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    // TEMP - Test suite failed to run
    //      TypeError: Cannot read property 'randomBytes' of undefined
    //       > 5 | import {Keystore, IKeystore} from "@nodefactory/bls-keystore";
    //
    //
    // it("renders correctly, Validator simple", () => {
    //     const tree = renderer
    //         .create(<ValidatorSimple
    //             name={"Validator 002"}
    //             status={"Not working"}
    //             publicKey={"6ffa3d24c9c26877d4a8bfa87455f44666ce93b7e13a3f84"}
    //             deposit={20.345}
    //             onExportClick={(): void=>{}} 
    //             onRemoveClick={(): void=>{}} 
    //             privateKey="6ffb3d24c9c26877d4a8bfa87455f44666ce93b7e13a3f84"
    //         />)
    //         .toJSON();
    //     expect(tree).toMatchSnapshot();
    // });
});


