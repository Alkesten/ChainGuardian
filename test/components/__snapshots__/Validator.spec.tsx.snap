// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validator renders correctly, 2 nodes 1`] = `
<div
  className="validator-container"
>
  <div
    className="validator-stats"
  >
    <h2>
      Validator 
      Test title
    </h2>
    <div
      className="validator-stats-container"
    >
      <div
        className="validator-card-container"
      >
        <h5>
          Return (ETH)
        </h5>
        <h1
          className="plus"
        >
          +
          10
          %
        </h1>
        <h5
          className="plus"
        >
          ROI
        </h5>
      </div>
      <div
        className="validator-card-container"
      >
        <h5>
          Balance
        </h5>
        <h1>
          .1206
        </h1>
        <h5>
          ETH
        </h5>
      </div>
      <div
        className="validator-card-container"
      >
        <h5>
          Uptime
        </h5>
        <h1>
          92.1
        </h1>
        <h5>
          DAYS
        </h5>
      </div>
    </div>
  </div>
  <div
    className="validator-nodes"
  >
    <div
      className="node-container"
    >
      <div
        className="node-grid-container"
      >
        <div
          className="node-card-container"
          onClick={[Function]}
        >
          <h2>
            BeaconNode1
          </h2>
          <span
            className="node-text"
          >
            rocketsonic1.hr
          </span>
          <h2>
            21
            ms
          </h2>
          <span
            className="node-text time"
          >
            Average Response Time
          </span>
        </div>
        <div
          className="node-card-container"
          onClick={[Function]}
        >
          <h2>
            BeaconNode2
          </h2>
          <span
            className="node-text"
          >
            rocketsonic2.hr
          </span>
          <h2>
            22
            ms
          </h2>
          <span
            className="node-text time"
          >
            Average Response Time
          </span>
        </div>
      </div>
      <button
        className="add-button"
        onClick={[Function]}
      >
        <img
          className="icon"
          src="test-file-stub"
        />
      </button>
      <div
        className="show 
                            none
                            more"
        onClick={[Function]}
      >
        Show More
      </div>
    </div>
    <div
      className="validator-buttons"
    >
      <button
        className="button btn-destructive "
        onClick={[Function]}
      >
        <div
          className=""
        >
          REMOVE
        </div>
      </button>
      <button
        className="button btn-primary "
        onClick={[Function]}
      >
        <div
          className=""
        >
          DETAILS
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`Validator renders correctly, 4 nodes 1`] = `
<div
  className="validator-container"
>
  <div
    className="validator-stats"
  >
    <h2>
      Validator 
      Test title
    </h2>
    <div
      className="validator-stats-container"
    >
      <div
        className="validator-card-container"
      >
        <h5>
          Return (ETH)
        </h5>
        <h1
          className="plus"
        >
          +
          10
          %
        </h1>
        <h5
          className="plus"
        >
          ROI
        </h5>
      </div>
      <div
        className="validator-card-container"
      >
        <h5>
          Balance
        </h5>
        <h1>
          .1206
        </h1>
        <h5>
          ETH
        </h5>
      </div>
      <div
        className="validator-card-container"
      >
        <h5>
          Uptime
        </h5>
        <h1>
          92.1
        </h1>
        <h5>
          DAYS
        </h5>
      </div>
    </div>
  </div>
  <div
    className="validator-nodes"
  >
    <div
      className="node-container"
    >
      <div
        className="node-grid-container"
      >
        <div
          className="node-card-container"
          onClick={[Function]}
        >
          <h2>
            BeaconNode1
          </h2>
          <span
            className="node-text"
          >
            rocketsonic1.hr
          </span>
          <h2>
            21
            ms
          </h2>
          <span
            className="node-text time"
          >
            Average Response Time
          </span>
        </div>
        <div
          className="node-card-container"
          onClick={[Function]}
        >
          <h2>
            BeaconNode2
          </h2>
          <span
            className="node-text"
          >
            rocketsonic2.hr
          </span>
          <h2>
            22
            ms
          </h2>
          <span
            className="node-text time"
          >
            Average Response Time
          </span>
        </div>
      </div>
      <button
        className="add-button"
        onClick={[Function]}
      >
        <img
          className="icon"
          src="test-file-stub"
        />
      </button>
      <div
        className="show 
                            
                            more"
        onClick={[Function]}
      >
        Show More
      </div>
    </div>
    <div
      className="validator-buttons"
    >
      <button
        className="button btn-destructive "
        onClick={[Function]}
      >
        <div
          className=""
        >
          REMOVE
        </div>
      </button>
      <button
        className="button btn-primary "
        onClick={[Function]}
      >
        <div
          className=""
        >
          DETAILS
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`Validator renders correctly, Validator simple 1`] = `
<div
  className="validator-container"
>
  <div
    className="validator-stats"
  >
    <h2>
      Validator 002
    </h2>
    <div
      className="validator-stats-container"
    >
      <div
        className="validator-card-container"
      >
        <h5>
          Deposit
        </h5>
        <h1>
          20.345
        </h1>
        <h5>
          ETH
        </h5>
      </div>
    </div>
    <h5
      className="public-key"
    >
      PUBLIC KEY: 
      6ffa3d24c9c26877d4a8bfa87455f44666ce93b7e13a3f84
    </h5>
  </div>
  <div
    className="validator-nodes-simple"
  >
    <h2>
      Status: 
      Not working
    </h2>
    <div
      className="validator-buttons-simple"
    >
      <button
        className="button btn-destructive "
        onClick={[Function]}
      >
        <div
          className=""
        >
          REMOVE
        </div>
      </button>
      <button
        className="button btn-secondary "
        onClick={[Function]}
      >
        <div
          className=""
        >
          EXPORT
        </div>
      </button>
    </div>
  </div>
</div>
`;
